src/hooks.ts

```diff
@@ -0,0 +1,45 @@
+import * as webpack from 'webpack';
+import { AsyncSeriesHook, SyncHook } from 'tapable';
+
+const compilerHookMap = new WeakMap();
+
+export const customHooks = {
+    forkTsCheckerServiceBeforeStart: 'fork-ts-checker-service-before-start',
+    forkTsCheckerCancel: 'fork-ts-checker-cancel',
+    forkTsCheckerServiceStartError: 'fork-ts-checker-service-start-error',
+    forkTsCheckerWaiting: 'fork-ts-checker-waiting',
+    forkTsCheckerServiceStart: 'fork-ts-checker-service-start',
+    forkTsCheckerReceive: 'fork-ts-checker-receive',
+    forkTsCheckerServiceOutOfMemory: 'fork-ts-checker-service-out-of-memory',
+    forkTsCheckerEmit: 'fork-ts-checker-emit',
+    forkTsCheckerDone: 'fork-ts-checker-done',
+};
+
+function createForkTsCheckerWebpackPluginHooks(): Record<keyof typeof customHooks, SyncHook | AsyncSeriesHook> {
+    return {
+        forkTsCheckerServiceBeforeStart: new AsyncSeriesHook([]),
```

Comment:
You could use shorter names now, as it's already scoped to this plugin:

```suggestion
        serviceBeforeStart: new AsyncSeriesHook([]),
```